modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/TableKeyDetails/index.js

```diff
@@ -0,0 +1,300 @@
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayIcon from '@clayui/icon';
+import classNames from 'classnames';
+import getCurrentEndDate from '../../../../common/utils/getCurrentEndDate';
+import {
+	getFormatedProductName,
+	getProductDescription,
+	getProductName,
+	getStatusActivationTag,
+	hasVirtualCluster,
+} from '../../containers/DXPActivationKeysTable/utils/constants';
+
+const HOST_NAME = 'Host Name';
+const IP_ADDRESSES = 'IP Addresses';
+const MAC_ADDRESSES = 'Mac Addresses';
+const SUBSCRIPTION_IMAGE_FILE = 'dxp_icon.svg';
+
+const NO_EXPIRATION_DATE = 100;
+
+const TableKeyDetails = ({
+	activationKeys,
+	assetsPath,
+	setValueToCopyToClipboard,
+}) => {
+	const instanceSizeFormated = activationKeys.sizing.slice(7, 8);
+
+	const unlimitedLicenseDate = new Date().setFullYear(
+		new Date().getFullYear() + NO_EXPIRATION_DATE
+	);
+
+	const handleExpiredDate =
+		new Date(activationKeys.expirationDate) >=
+		new Date(unlimitedLicenseDate)
+			? 'Does Not Expire'
+			: getCurrentEndDate(activationKeys.expirationDate);
+
+	const handleCopyToClipboard = (value, action) => {
+		navigator.clipboard.writeText(action);
+
+		setValueToCopyToClipboard(value);
+	};
+
+	return (
+		<>
+			<div className="container">
+				<div className="row">
+					<div className="col-5">
+						<h5>Environment</h5>
+					</div>
+
+					<div className="col-4">
+						<h5>Server</h5>
+					</div>
+
+					<div className="col-3">
+						<h5>Activation Status</h5>
+					</div>
+				</div>
+
+				<div className="row">
+					<div className="col-2">
+						<p className="text-neutral-8 text-paragraph-sm">
+							Product
+						</p>
+					</div>
+
+					<div className="col-3">
+						<p className="text-neutral-8 text-paragraph-sm">
+							Version
+						</p>
+					</div>
+
+					<div className="col-4">
+						<p className="text-neutral-8 text-paragraph-sm">
+							Key Type
+						</p>
+					</div>
+
+					<div className="col-3">
+						<p className="text-neutral-8 text-paragraph-sm">
+							Status
+						</p>
+					</div>
+				</div>
+
+				<div className="row">
+					<div className="col-2">
+						<p className="align-items-center bg-brand-primary-lighten-5 d-flex key-details-paragraph px-3 py-2 rounded">
+							<img
+								className="mr-2"
+								src={`${assetsPath}/assets/navigation-menu/${SUBSCRIPTION_IMAGE_FILE}`}
+							/>
+
+							{getFormatedProductName(activationKeys)}
+						</p>
+					</div>
+
+					<div className="col-3">
+						<p className="bg-brand-primary-lighten-5 key-details-paragraph px-3 py-2 rounded">
+							{activationKeys.productVersion}
+						</p>
+					</div>
+
+					<div className="col-4">
+						<p className="bg-neutral-1 key-details-paragraph px-3 py-2 rounded">
+							{hasVirtualCluster(activationKeys)
```

Comment:
I noticed that you use this function many times, should be good pass the value for a variable before to avoid run many times.

`const hasVirtualClusterForActivationKeys = hasVirtualCluster(activationKeys);`

Should be good do this for all functions, create the variable before and just use it to avoid re-run the same function.

`const statusActivationTag = getStatusActivationTag(activationKeys)`

```suggestion
							{hasVirtualClusterForActivationKeys
```